fusesoc/main.py

```diff
@@ -155,9 +156,19 @@ def add_library(cm, args):
     library = {}
     name = args.name
     sync_uri = vars(args)['sync-uri']
-    if os.path.isdir(sync_uri) and not args.location:
-        logger.info("Detecting {} as a local library".format(sync_uri))
+    if 'sync-type' in vars(args):
+        provider = vars(args)['sync-type']
+        library['sync-type'] = provider
+    else:
+        provider = 'git'
+
+    if provider == 'local' and not args.location:
+        logger.info("Interpreting sync-uri '{}' as location for local provider.".format(sync_uri))
         library['location'] = os.path.abspath(sync_uri)
+        library['sync-type'] = 'local'
+    elif provider == 'local' and os.path.abspath(args.location) != os.path.abspath(sync_uri):
+        logger.error("Location '{}' does not match sync-uri '{}' for local provider. Consider specifying only sync-uri.".format(args.location, sync_uri))
+        exit(1)
```

Comment:
Wouldn't it make more sense to skip sync-uri and only use location for local libraries?

Reply:
If we do this, then we'll break #191 - `fusesoc library add dummy ../Desktop` won't work. This was the best compromise I found.

Reply:
I see. It's unfortunate that sync-uri means different things for local and remote libraries. Maybe we need to go back to the drawing table and rethink the CLI slightly, but until then we do it like this

Reply:
The issue is somewhat fundamental, I think. For all other provider types, a sync-uri of some sort is required. Local is always going to be special.

Perhaps we could do `fusesoc library add git http://github.com/fusesoc/fusesoc-cores` and `fusesoc library add local ../Desktop`, making the sync-type a required positional argument? This would allow each provider to take different arguments, which could be good or bad.